### DISCUSSION TITLE
Committing to release branches

### DISCUSSION THREAD

**Discussion by rcomer** [SciTools/iris#4996] [Ideas]
I've noticed that some projects target all their contributor PRs at `main` and have a bot that then, if necessary, reproduces the change on a release branch.  For example, for this recent PR in pytest

* https://github.com/pytest-dev/pytest/pull/10301

the reviewer merges the PR to main and adds the `backport 7.1.x` label.  The GitHub Action then comes along, sees the label, and opens a new PR with the same change against the 7.1.x branch

* https://github.com/pytest-dev/pytest/pull/10302

Matplotlib does something similar, but with some sort of external bot, and based on the milestone. e.g.
* https://github.com/matplotlib/matplotlib/pull/23964
* https://github.com/matplotlib/matplotlib/pull/23994

Advantages to this approach:

* No confusion over which branch a contributor should target.
* Possibly less hassle with mergeback PRs (I've not been involved in those, but they don't look much fun...)
* Since bugfixes are merged straight into main, any linked issues will automatically close.

**Comment by trexfeathers** [comment]
Interesting. Bits of this model might also help with @abooton's suggestion of duplicate PR's for feature branches too.

**Comment by rcomer** [comment]
Relevant parts of those projects’ dev guides:

- [pytest](https://github.com/pytest-dev/pytest/blob/main/CONTRIBUTING.rst#backporting-bug-fixes-for-the-next-patch-release)
- [matplotlib](https://matplotlib.org/stable/devel/coding_guide.html#branches-and-backports)
